integration/setup/setup.go

```diff
@@ -125,6 +125,7 @@ func Setup(m *testing.M, config Config) {
 			v = 1
 		}
 
+		// TODO discuss how useful this is at all. Who needs this and why?
 		if !env.CircleCI() {
 			err := teardown(ctx, config)
```

Comment:
Question for everyone. I feel like this is not used at this point.

Reply:
The only reason I can see for it is clearing up leftover resources from failed e2e tests. If that is covered then I'm fine with removing.

I think it might have been to help when running locally with `minikube`. But with `kind` I think a fresh container per local test is the correct workflow.

Reply:
We do not have anything like cleaning up resources from failed tests. The condition above is also only true when not executed in Circle CI, which means executing locally. So anyone running the test locally might want this. I never did that though.

Reply:
I was thinking of https://github.com/giantswarm/ci-cleaner that cleans up AWS resources. I think that is still used?

Reply:
That is in use, but it is not complete. The CI cleaner only purges CF stacks and S3 buckets. The teardown here is to cleanup helm releases from the e2e control plane. But again, we do not run the teardown in Circle.

Reply:
I dropped that code. When anyone has something against it, let me know.